plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftPublishController.java

```diff
@@ -42,22 +39,20 @@
 
 public class OpenShiftPublishController extends StandardFileSystemPublishController implements IPublishController {
 
-	private RSync rsync = null;
-	private RSync createRSync(IServer server, IProgressMonitor monitor) throws CoreException {
-		String location = OCBinary.getInstance().getLocation();
+	private static RSync createRSync(IServer server, IProgressMonitor monitor) throws CoreException {
+		final String location = OCBinary.getInstance().getLocation();
```

Comment:
I feel like this should be checked in RSync (or even better in any OCBinaryOperation) so that we dont have to repeat ourselves.